logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java

```diff
@@ -30,13 +31,24 @@ public static HeaderFilter replaceHeaders(final BiPredicate<String, String> pred
         return eachHeader((key, value) -> predicate.test(key, value) ? replacement : value);
     }
 
-    public static HeaderFilter eachHeader(final BinaryOperator<String> operator) {
+    public static HeaderFilter removeHeaders(final Predicate<String> keyPredicate) {
+        return eachHeader((key, value) -> keyPredicate.test(key) ? null : value);
+    }
+
+    public static HeaderFilter removeHeaders(final BiPredicate<String, String> predicate) {
+        return eachHeader((key, value) -> predicate.test(key, value) ? null : value);
+    }
+
+    public static HeaderFilter eachHeader(final BiFunction<String, String, String> transform) {
```

Comment:
This should be a safe change since `BinaryOperator extends BiFunction`, but now it will remove value if `transform` returns `null`